inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/hook/MethodHook.java

```diff
@@ -0,0 +1,70 @@
+package rocks.inspectit.oce.core.instrumentation.hook;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import rocks.inspectit.oce.bootstrap.context.IInspectitContext;
+import rocks.inspectit.oce.bootstrap.instrumentation.IMethodHook;
+import rocks.inspectit.oce.core.instrumentation.config.model.MethodHookConfiguration;
+import rocks.inspectit.oce.core.instrumentation.context.ContextManager;
+
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+@Slf4j
+@Data
+@Builder
+public class MethodHook implements IMethodHook {
```

Comment:
this is also an important class: javadocs, small inline explanations where applicable..